### PR TITLE
Validate attributes in update methods

### PR DESCRIPTION
**Pull request by segments-arnaud**

### CODE REVIEW

File: src/segments/client.py [hunk 1]
```diff
@@ -1023,16 +1030,16 @@ def add_sample(
             :exc:`~segments.exceptions.TimeoutError`: If the request times out.
         """
 
-        if type(attributes) is dict:
+        if any(isinstance(attributes, Type) for Type in get_args(SampleAttributes)):
```
Comment (dbbert) [inline src/segments/client.py:1033]: Why did you change this? Your previous `if type(attributes) in get_args(SampleAttributes):` looks cleaner, or is it not the same?

Comment (segments-arnaud) [reply src/segments/client.py]: - It's better practise to use `isinstance` over `type`. When you have a class hierarchy Ball, Football(Ball) then type is too restrictive. It only works for the exact same type. `football = Football()` and `type(football) is Ball` is false. isinstance(ball, Ball) is true. So in summary, `isinstance` allows inheritance, `type` does not. 
- isinstance works with a collection of classes I just found out. So I can write it cleaner (remove the for loop and any) -> isinstance(attributes, get_args(SampleAttributes)).
- On a practical note, for our code it doesn't matter because we currently don't use inheritance for sample attributes and label attributes but use a union. In the future we may want to refactor this to use inheritance.

**Review by dbbert — APPROVED** [review]
